### PR TITLE
Delivery Rules & Delivery Fifo

### PR DESCRIPTION
**Pull request by av-ps**
delivery rules&fifo

### CODE REVIEW

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/C_BPartner_StepDef.java [hunk 1]
```diff
@@ -227,7 +228,9 @@ private void createC_BPartner(@NonNull final Map<String, String> tableRow, final
 			bPartnerRecord.setInvoiceRule(invoiceRule);
 		}
 
-		bPartnerRecord.setDeliveryRule(DELIVERYRULE_Force);
+		final String deliveryRule = DataTableUtil.extractStringOrNullForColumnName(tableRow, "OPT." + COLUMNNAME_DeliveryRule);
+		bPartnerRecord.setDeliveryRule(Check.isNotBlank(deliveryRule) ? deliveryRule : DELIVERYRULE_Force);
```
Comment (metas-ts) [inline backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/C_BPartner_StepDef.java:232]: here you could also use `de.metas.common.util.CoalesceUtil#firstNotBlank(java.lang.String...)` ..just as hint for next time :-)

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/hu/M_HU_StepDef.java [hunk 2]
```diff
@@ -237,10 +238,13 @@ public void find_HUs(final int timeoutSec, @NonNull final DataTable dataTable) t
 		for (final Map<String, String> tableRow : dataTable.asMaps())
 		{
 			final String inventoryLineIdentifier = DataTableUtil.extractStringForColumnName(tableRow, I_M_InventoryLine.COLUMNNAME_M_InventoryLine_ID + "." + TABLECOLUMN_IDENTIFIER);
+			final Integer inventoryLineId = inventoryLineTable.getOptional(inventoryLineIdentifier)
+					.map(I_M_InventoryLine::getM_InventoryLine_ID)
+					.orElseGet(() -> Integer.parseInt(inventoryLineIdentifier));
```
Comment (metas-ts) [inline backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/hu/M_HU_StepDef.java:243]: from a quick search for `s, there are added M_HUs for inventory` among feature files, we can AFAIU drop
the fallback to `Integer.parseInt(inventoryLineIdentifier));`. everyone just uses the actual identifier

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/shipment/M_InOut_StepDef.java [hunk 3]
```diff
@@ -355,6 +363,20 @@ public void locate_shipment_by_scheduleId(@NonNull final DataTable table)
...
+	public void no_M_InOut_found(@NonNull final String orderIdentifier)
+	{
+		final I_C_Order order = orderTable.get(orderIdentifier);
+		assertThat(order).isNotNull();
+
+		final I_M_InOut inOut = queryBL.createQueryBuilder(I_M_InOut.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(COLUMNNAME_C_Order_ID, order.getC_Order_ID())
+				.create()
+				.firstOnlyOrNull(I_M_InOut.class);
```
Comment (metas-ts) [inline backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/shipment/M_InOut_StepDef.java:376]: 🤔 AFAIU if there are two M_InOuts, then this step will succeed although it should fail

File: backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/creditLimit.feature [hunk 4]
```diff
@@ -87,7 +87,7 @@ Feature: credit limit
 
     And the metasfresh REST-API endpoint path 'api/v2/shipments' receives a 'POST' request with the payload from context and responds with '200' status code
 
-    Then validate created M_ShipmentSchedule_QtyPicked records
+    Then validate single  M_ShipmentSchedule_QtyPicked record created for shipment schedule
```
Comment (metas-ts) [inline backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/creditLimit.feature:90]: one space too much

**Review by metas-ts — CHANGES_REQUESTED** [review]
..minor requests

File: backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/shipment/M_InOut_StepDef.java (hunk 3 above)
Comment (cp-ps) [reply backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/shipment/M_InOut_StepDef.java:376]: 🤯 right -> I've looked at it as a firstOnlyOptional

**Pull request merged**